warehouse/query-core/src/main/java/datawave/query/iterator/logic/TermFrequencyIndexIterator.java

```diff
@@ -187,12 +187,20 @@ public void next() throws IOException {
             
             for (int i = 0; i < 256 && source.hasTop() && key.getFieldName().compareTo(field) < 0; ++i) {
                 source.next();
-                key = new DatawaveKey(source.getTopKey());
+                
+                tk = source.getTopKey();
```

Comment:
Reusing tk here is not good here.  Say we get through this loop without tk being null, and then the check at line 204 is true.  This would result in a seek and we would immediately pop out of the outer while loop because tk is not null and would not be holding the correct field name!  Use a separate Key variable instead.

Reply:
Will do